test/discretenonparametric.jl

```diff
@@ -112,4 +112,14 @@ p = [1 - eps(Float32), eps(Float32)]
 d = Categorical(p)
 @test ([rand(d) for _ = 1:100_000]; true)
 
+# Numerical stability w/ large prob vectors;
+# see issue #1017
+n = 20000 # large vector length
+p = Float32[0.5; fill(0.5/(n ÷ 2) - 3e-8, n ÷ 2); fill(eps(Float64), n ÷ 2)]
+d = Categorical(p)
+struct DumbRNG <: AbstractRNG end
```

Comment:
The DumbRNG definition and method implementation should be at the top-level scope, outside the testset macro (see Travis error on 1.0)